app/models/residence.rb

```diff
@@ -33,8 +33,12 @@ def initialize_residence
   end
 
   def to_s
-    if current
+    if current && /^homeless/i.match(address.to_s)
+     "Currently homeless"
+    elsif current
       "Current residence at #{address}"
+    elsif /homeless/i.match(address.to_s)
```

Comment:
The "homeless?" Method probably belongs in the Address model, since you have a regex there too.